### DISCUSSION TITLE
Country-Codes from RiRData - No login and wrong country because of IPv6?

### DISCUSSION THREAD

**Discussion by MichaelS844** [HomeITAdmin/nextcloud_geoblocker#68] [Q&A]
Hello, 
at first I like to thank for for this app! really useful and handy. 
Unfortunately, if I activate the app, I can't login any more. 
I'm using RIRData with an up-to-date database, I whitelisted "DE". In php.ini "allow_url_fopen" is set to true. 

But if I try to login from DE, I get blocked: 
`{... The user \"NOT_SHOWN_IN_LOG\" attempt to login with IP address \"2a03:.......\" from blocked country \"*CZ*\". Login is blocked by exception.","userAgent":"....
`
If I use another device, I get blocked with the same message, but the country in the message changes to "AA"

So, I'm using ipv6 on the server and on my other devices. Could this cause the problem and does anyone know how to fix this? 

Thanks for your response.

**Comment by HomeITAdmin** [comment]
Basically IPv6 should also work. But maybe I have still a bug in the implementation (more likely). Another option would be, that the RIR data is wrong (less likely).
The RIR data for Europe is here:  https://ftp.ripe.net/pub/stats/ripencc/delegated-ripencc-latest
As you can see, for the beginning "2a03:..." this could be a lot of different countries depending on how it goes on the next elements. Can you share the next four elements with me, so that I can check, what the expected result should have been on the basis of that data? Can you also share the beginning of the second IP with the other device?

**Reply by MichaelS844** [reply]
Thanks for your answer!
Yes of course, here are the next 4 characters and the second device.

Device 1:
`The user \"NOT_SHOWN_IN_LOG\" attempt to login with IP address \"2a03:567:....\" from blocked country \"CZ\".`
Device 2:
`{... The user \"NOT_SHOWN_IN_LOG\" attempt to login with IP address \"2a02:8070:........." from blocked country \"AA\".`

I checked them against de RIR Data from your link. The result should be "DE", I think.
Device 1: ripencc | DE | ipv6 |2a03:560:: | 29| 20140701| allocated
Device 2: ripencc | DE | ipv6 |2a02:8070:: | 31| 20110913| allocated

By the way, I switched to Geoiplookup-Method and this works perfectly ;)

**Reply by HomeITAdmin** [reply]
Yes, you are right, so the error must be in my code. 
I created the issue #69 to track the fix.

**Reply by HomeITAdmin** [reply]
But good the the Geoiplookup Method is working ;-)
